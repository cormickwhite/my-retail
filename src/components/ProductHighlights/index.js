import React, { Component } from 'react';

class ProductHighlights extends Component {
  render() {
    const { highlights } = this.props;
    return (
      <div className="component">
        <h1>product highlights</h1>
        {highlights.map((highlight, index) => {
          return (
            <li key={index} dangerouslySetInnerHTML={{ __html: highlight }} />
          );
        })}
      </div>
    );
  }
}

export default ProductHighlights;
