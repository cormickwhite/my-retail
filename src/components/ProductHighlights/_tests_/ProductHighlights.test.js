import React from 'react';
import { shallow } from 'enzyme';
import ProductHighlights from '../index.js';

it('renders html features', () => {
  const features = [
    '<strong>Painter:</strong> Bob Ross',
    '<strong>Brushes Included:</strong> 3 ',
    '<strong>Paints Included:</strong> 7',
    '<strong>Includes:</strong> Paint'
  ];

  const wrapper = shallow(<ProductHighlights highlights={features} />);

  expect(wrapper).toMatchSnapshot();
});
