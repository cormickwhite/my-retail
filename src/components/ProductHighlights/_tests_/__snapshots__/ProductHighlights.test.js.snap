// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders html features 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProductHighlights
    highlights={
        Array [
            "<strong>Painter:</strong> Bob Ross",
            "<strong>Brushes Included:</strong> 3 ",
            "<strong>Paints Included:</strong> 7",
            "<strong>Includes:</strong> Paint",
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          product highlights
</h1>,
        Array [
          <li
            dangerouslySetInnerHTML={
                        Object {
                                    "__html": "<strong>Painter:</strong> Bob Ross",
                                  }
            }
/>,
          <li
            dangerouslySetInnerHTML={
                        Object {
                                    "__html": "<strong>Brushes Included:</strong> 3 ",
                                  }
            }
/>,
          <li
            dangerouslySetInnerHTML={
                        Object {
                                    "__html": "<strong>Paints Included:</strong> 7",
                                  }
            }
/>,
          <li
            dangerouslySetInnerHTML={
                        Object {
                                    "__html": "<strong>Includes:</strong> Paint",
                                  }
            }
/>,
        ],
      ],
      "className": "component",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "product highlights",
        },
        "ref": null,
        "rendered": "product highlights",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "dangerouslySetInnerHTML": Object {
            "__html": "<strong>Painter:</strong> Bob Ross",
          },
        },
        "ref": null,
        "rendered": null,
        "type": "li",
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "dangerouslySetInnerHTML": Object {
            "__html": "<strong>Brushes Included:</strong> 3 ",
          },
        },
        "ref": null,
        "rendered": null,
        "type": "li",
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "host",
        "props": Object {
          "dangerouslySetInnerHTML": Object {
            "__html": "<strong>Paints Included:</strong> 7",
          },
        },
        "ref": null,
        "rendered": null,
        "type": "li",
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "host",
        "props": Object {
          "dangerouslySetInnerHTML": Object {
            "__html": "<strong>Includes:</strong> Paint",
          },
        },
        "ref": null,
        "rendered": null,
        "type": "li",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            product highlights
</h1>,
          Array [
            <li
              dangerouslySetInnerHTML={
                            Object {
                                          "__html": "<strong>Painter:</strong> Bob Ross",
                                        }
              }
/>,
            <li
              dangerouslySetInnerHTML={
                            Object {
                                          "__html": "<strong>Brushes Included:</strong> 3 ",
                                        }
              }
/>,
            <li
              dangerouslySetInnerHTML={
                            Object {
                                          "__html": "<strong>Paints Included:</strong> 7",
                                        }
              }
/>,
            <li
              dangerouslySetInnerHTML={
                            Object {
                                          "__html": "<strong>Includes:</strong> Paint",
                                        }
              }
/>,
          ],
        ],
        "className": "component",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "product highlights",
          },
          "ref": null,
          "rendered": "product highlights",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "dangerouslySetInnerHTML": Object {
              "__html": "<strong>Painter:</strong> Bob Ross",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "dangerouslySetInnerHTML": Object {
              "__html": "<strong>Brushes Included:</strong> 3 ",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "dangerouslySetInnerHTML": Object {
              "__html": "<strong>Paints Included:</strong> 7",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "host",
          "props": Object {
            "dangerouslySetInnerHTML": Object {
              "__html": "<strong>Includes:</strong> Paint",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "li",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
